packages/docs/src/stories/styles/pagination.test.stories.ts

```diff
@@ -0,0 +1,255 @@
+    ...parameters,
+    controls: { disable: true },
+    design: {
+      type: 'figma',
+      url: 'https://www.figma.com/design/VTztxQ5pWG7ARg8hCX6PfR/branch/LiEAPa5PkSJyjkyAE9RNsi/Solid-DS-%E2%80%93-Component-Library?node-id=9401-33&p=f&t=A7lFFTs8C5yV9hmi-0'
+    }
+  },
+  args: overrideArgs([
+    {
+      type: 'slot',
+      name: 'default',
+      value: html`
+      <h2>Pagination</h2>
+      <ul>
+        <li><a aria-hidden="true"><sd-icon name="system/chevron-left" label="Previous"></a></li>
+        <li><a href="/?page=1" data-page="1" aria-current="page">1</a></li>
+        <li><a href="/?page=2" data-page="2">2</a></li>
+        <li><a href="/?page=3" data-page="3">3</a></li>
+        <li><a href="/?page=4" data-page="4">4</a></li>
+        <li><a href="/?page=5" data-page="5">5</a></li>
+        <li><a href="/?page=2"><sd-icon name="system/chevron-right" label="Next"></a></li>
+      </ul>
+    `
+    }
+  ]),
+  argTypes,
+  decorators: [withActions] as any
+};
+
+export const Default = {
+  name: 'Default',
+  render: (args: any) => {
+    return generateTemplate({
+      options: {
+        templateContent: html`<nav class="%CLASSES%" aria-label="Pagination">%SLOT%</nav> `
+      },
+      args
+    });
+  }
+};
+
+export const Variants = {
+  name: 'Variants',
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: [{ type: 'attribute', name: 'sd-pagination--simple', values: [false, true] }]
+      },
+      options: {
+        templateContent: html`
+          <nav class="%CLASSES%" aria-label="Pagination">
+            <ul>
+              <li><a aria-hidden="true"><sd-icon name="system/chevron-left" label="Previous"></a></li>
+              <li><a href="/?page=1" data-page="1" aria-current="page">1</a></li>
+              <li><a href="/?page=2" data-page="2">2</a></li>
+              <li><a href="/?page=3" data-page="3">3</a></li>
+              <li><a href="/?page=4" data-page="4">4</a></li>
+              <li><a href="/?page=5" data-page="5">5</a></li> 
+              <li><a href="/?page=2"><sd-icon name="system/chevron-right" label="Next"></a></li>
+            </ul>
+          </nav>
+        `
+      },
+      args
+    });
+  }
+};
+
+export const SinglePage = {
+  name: 'Single Page',
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: [{ type: 'attribute', name: 'sd-pagination--simple', values: [false, true] }]
+      },
+      options: {
+        templateContent: html`
+          <nav class="%CLASSES%" aria-label="Pagination">
+            <ul>
+              <li><a aria-hidden="true"><sd-icon name="system/chevron-left" label="Previous"></a></li>
+              <li><a href="/?page=1" data-page="1" aria-current="page">1</a></li>
+              <li><a aria-hidden="true"><sd-icon name="system/chevron-right" label="Next"></a></li>
+            </ul>
+          </nav>
+        `
+      },
+      args
+    });
+  }
+};
+
+export const Truncated = {
```

Comment:
Please add all the tests from:

https://codepen.io/mariohamann/pen/emYzbZP?editors=1110

There are some edges cases, that should be reflected in screenshot tests as well.